lib/core/test/unit/Cardano/Wallet/DB/SqliteSpec.hs

```diff
@@ -177,13 +177,12 @@ spec = do
     sqliteDetailedSeqSpec
 
 sqliteDetailedSeqSpec :: Spec
-sqliteDetailedSeqSpec = withDB
-    (fst <$> newFileDBLayer @(SeqState DummyTarget) @DummyTarget @SeqKey) $ do
+sqliteDetailedSeqSpec = withDB newDBSeq $ do
```

Comment:
@rvl unit tests will fail with this change.  I have used fileBased approach as otherwise in in-memory solution I cannot use SQL side by side. newDBSeq relies on in-memory solution, and to my best knowledge you cannot use runSqlite' with it - it just instantiates next instance of db with no schema and we will end up with : 
```
est/unit/Cardano/Wallet/DB/SqliteSpec.hs:212:9: 
  1) Cardano.Wallet.DB.Sqlite, Sqlite multiple-checkpoints test (SeqState), put and read checkpoints, previous checkpoints are also stored
       uncaught exception: SqliteException
       SQLite3 returned ErrorError while attempting to perform prepare "SELECT \"wallet_id\", \"slot\", \"parent_block_id\", \"block_height\" FROM \"checkpoint\"": no such table: checkpoint
```
Of course, it would be much better if the tests only used DBLayer functions. But....I needed them to make sure every step is reflected in DB tables as expected. I wanted to follow the life-cycle of utxo row, make sure they are not superfluous rows, read spent_utxo column, ... those things are digested by DBlayer methods. At this moment only inspecting DBlayer methods side by side with SQL SELECT I am confident I can follow what exactly happens in DB - hence the tests. But I agree, using DBlayer methods would considerably simplify tests (we could use in-memory solution for example)

Reply:
now unit tests should pass in new commit https://github.com/input-output-hk/cardano-wallet/pull/686/commits/380fd075f17a83dbca25d89f5f3048edceb97858

Reply:
Oops, I didn't see that it was a different `newDB` function.
There is:
```
unsafeRunQuery :: SqliteContext -> SqlPersistT IO a -> IO a
```
Exported from the Sqlite module and you could add a test function:
```
withDBCtx :: SpecWith (SqliteContext, DBLayer IO s t k) -> Spec
withDBCtx = beforeAll newMemoryDBLayer . beforeWith (\(ctx, db) -> cleanDB db $> (ctx, db))
```
That should allow you to run arbitrary queries, on a memory database, with migrations and logging.

Reply:
done. thanks to that both RndState and SeqState was embraced